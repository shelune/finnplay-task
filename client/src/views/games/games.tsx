import classNames from "classnames";
import React, { FC, useCallback, useMemo, useState } from "react";

import { ReactComponent as SearchIcon } from "../../assets/icons/search.svg";
import { ReactComponent as HamburgerIcon } from "../../assets/icons/hamburger.svg";
import css from "./games.module.scss";
import { LocalGame, LocalGameGroup, LocalProvider } from "../../utils/types";

type Props = {
  games: LocalGame[];
  providers: LocalProvider[];
  groups: LocalGameGroup[];
};

type SortingType = "A-Z" | "Z-A" | "Newest";

const COLUMN_OPTIONS = [2, 3, 4];

export const GamesView: FC<Props> = ({ games, providers, groups }) => {
  const [search, setSearch] = useState("");
  const [activeProviders, setActiveProviders] = useState<number[]>([]);
  const [activeGroups, setActiveGroups] = useState<number[]>([]);
  const [sorting, setSorting] = useState<SortingType>("A-Z");
  const [columnCount, setColumnCount] = useState(4);
  const [showFilters, setShowFilters] = useState(false);

  const filteredGames = useMemo(() => {
    const filtered = games
      .filter((game) => {
        if (search) {
          return game.name.toLowerCase().includes(search.toLowerCase());
        }
        return game;
      })
      .filter(
        (game) =>
          activeProviders.includes(game.provider) ||
          activeProviders.length === 0,
      )
      .filter((game) => {
        if (activeGroups.length === 0) {
          return true;
        }
        const activeGroupsFull = groups.filter((group) =>
          activeGroups.includes(group.id),
        );
        const gamesIdInActiveGroups = activeGroupsFull.reduce((accu, curr) => {
          const { games: gamesOfGroup } = curr;
          return [...accu, ...gamesOfGroup];
        }, [] as number[]);
        return gamesIdInActiveGroups.includes(game.id);
      });

    const filteredSort = filtered.sort((a, b) => {
      if (sorting === "A-Z") {
        return a.name.toLowerCase().localeCompare(b.name.toLowerCase());
      }
      if (sorting === "Z-A") {
        return b.name.toLowerCase().localeCompare(a.name.toLowerCase());
      }
      if (sorting === "Newest") {
        return b.date.getTime() - a.date.getTime();
      }
      return 0;
    });
    return filteredSort;
  }, [activeGroups, activeProviders, games, groups, search, sorting]);

  const resetFilters = useCallback(() => {
    setSearch("");
    setActiveProviders([]);
    setActiveGroups([]);
    setSorting("A-Z");
    setColumnCount(4);
  }, []);

  return (
    <div className={css.gamesPage}>
      <div className={css.content}>
        <div
          className={classNames(css.gameFiltersMobile, {
            [css.isShowingFilters]: showFilters,
          })}
        >
          <div className={css.filterBySearch}>
            <div className={css.searchBar}>
              <input
                type="text"
                className={css.textInput}
                value={search}
                onChange={(e) => setSearch(e.target.value)}
                placeholder="Search"
              />
              <SearchIcon className={css.searchIcon} />
            </div>
          </div>
          {showFilters && (
            <>
              <div className={css.filterByProviders}>
                <div className={css.filterTitle}>Providers</div>
                <div className={css.filterList}>
                  {providers.map((provider) => (
                    <span
                      key={provider.id}
                      className={classNames(css.filterPill, {
                        [css.isActive]: activeProviders.includes(provider.id),
                      })}
                      role="button"
                      tabIndex={0}
                      onKeyDown={() => {}}
                      onClick={() => {
                        if (activeProviders.includes(provider.id)) {
                          setActiveProviders(
                            activeProviders.filter(
                              (item) => item !== provider.id,
                            ),
                          );
                        } else {
                          setActiveProviders([...activeProviders, provider.id]);
                        }
                      }}
                    >
                      {provider.name}
                    </span>
                  ))}
                </div>
              </div>
              <div className={css.filterByGroups}>
                <div className={css.filterTitle}>Game groups</div>
                <div className={css.filterList}>
                  {groups.map((group) => (
                    <span
                      key={group.id}
                      className={classNames(css.filterPill, {
                        [css.isActive]: activeGroups.includes(group.id),
                      })}
                      role="button"
                      tabIndex={0}
                      onKeyDown={() => {}}
                      onClick={() => {
                        if (activeGroups.includes(group.id)) {
                          setActiveGroups(
                            activeGroups.filter((item) => item !== group.id),
                          );
                        } else {
                          setActiveGroups([...activeGroups, group.id]);
                        }
                      }}
                    >
                      {group.name}
                    </span>
                  ))}
                </div>
              </div>
              <div className={css.sorting}>
                <div className={css.filterTitle}>Sorting</div>
                <div className={css.filterList}>
                  {["A-Z", "Z-A", "Newest"].map((item) => (
                    <span
                      key={item}
                      className={classNames(css.filterPill, {
                        [css.isActive]: sorting === item,
                      })}
                      role="button"
                      tabIndex={0}
                      onKeyDown={() => {}}
                      onClick={() => {
                        if (sorting !== item) {
                          setSorting(item as SortingType);
                        }
                      }}
                    >
                      {item}
                    </span>
                  ))}
                </div>
              </div>
              <div className={css.byline}>
                <div className={css.gamesCount}>
                  Game amount:
                  <span>{` ${filteredGames.length}`}</span>
                </div>
                <button
                  type="button"
                  className={css.resetFilters}
                  onClick={() => {
                    resetFilters();
                  }}
                >
                  Reset
                </button>
              </div>
            </>
          )}
          <div
            className={css.toggleFilters}
            role="button"
            tabIndex={0}
            onKeyDown={() => {}}
            onClick={() => setShowFilters(!showFilters)}
          >
            <HamburgerIcon className={css.toggleFiltersIcon} />
            <span>{showFilters ? "Hide filters" : "Show filters"}</span>
          </div>
        </div>
        {filteredGames.length ? (
          <div
            data-testid="game-list"
            className={classNames(css.gameList)}
            style={{
              gridTemplateColumns: `repeat(${columnCount}, 1fr)`,
            }}
          >
            {filteredGames.map((item) => (
              <div
                data-testid="game-box"
                key={item.id}
                title={item.name}
                className={css.gameBox}
                style={{
                  backgroundImage: `url(${item.cover})`,
                }}
              />
            ))}
          </div>
        ) : (
          <div className={css.gameList}>Sorry, no matching game found.</div>
        )}
        <div className={css.gameFilters}>
          <div className={css.filterBySearch}>
            <div className={css.searchBar}>
              <input
                data-testid="game-search-text"
                type="text"
                className={css.textInput}
                value={search}
                onChange={(e) => setSearch(e.target.value)}
                placeholder="Search"
              />
              <SearchIcon className={css.searchIcon} />
            </div>
          </div>
          <div className={css.filterByProviders}>
            <div className={css.filterTitle}>Providers</div>
            <div className={css.filterList}>
              {providers.map((provider) => (
                <span
                  key={provider.id}
                  className={classNames(css.filterPill, {
                    [css.isActive]: activeProviders.includes(provider.id),
                  })}
                  role="button"
                  tabIndex={0}
                  onKeyDown={() => {}}
                  onClick={() => {
                    if (activeProviders.includes(provider.id)) {
                      setActiveProviders(
                        activeProviders.filter((item) => item !== provider.id),
                      );
                    } else {
                      setActiveProviders([...activeProviders, provider.id]);
                    }
                  }}
                >
                  {provider.name}
                </span>
              ))}
            </div>
          </div>
          <div className={css.filterByGroups}>
            <div className={css.filterTitle}>Game groups</div>
            <div className={css.filterList}>
              {groups.map((group) => (
                <span
                  key={group.id}
                  className={classNames(css.filterPill, {
                    [css.isActive]: activeGroups.includes(group.id),
                  })}
                  role="button"
                  tabIndex={0}
                  onKeyDown={() => {}}
                  onClick={() => {
                    if (activeGroups.includes(group.id)) {
                      setActiveGroups(
                        activeGroups.filter((item) => item !== group.id),
                      );
                    } else {
                      setActiveGroups([...activeGroups, group.id]);
                    }
                  }}
                >
                  {group.name}
                </span>
              ))}
            </div>
          </div>
          <div className={css.sorting}>
            <div className={css.filterTitle}>Sorting</div>
            <div className={css.filterList}>
              {["A-Z", "Z-A", "Newest"].map((item) => (
                <span
                  key={item}
                  className={classNames(css.filterPill, {
                    [css.isActive]: sorting === item,
                  })}
                  role="button"
                  tabIndex={0}
                  onKeyDown={() => {}}
                  onClick={() => {
                    if (sorting !== item) {
                      setSorting(item as SortingType);
                    }
                  }}
                >
                  {item}
                </span>
              ))}
            </div>
          </div>
          <div className={css.filterByColumns}>
            <div className={css.filterTitle}>Columns</div>
            <div className={css.filterBar}>
              <div className={css.bar}>
                <div className={css.barBackground} />
                <div
                  className={css.barActive}
                  style={{
                    width: `calc(${
                      (100 *
                        COLUMN_OPTIONS.findIndex(
                          (option) => option === columnCount,
                        )) /
                      (COLUMN_OPTIONS.length - 1)
                    }%`,
                  }}
                />
              </div>
              {COLUMN_OPTIONS.map((count) => (
                <span
                  key={count}
                  className={classNames(css.milestone, {
                    [css.isActive]: columnCount >= count,
                  })}
                  role="button"
                  tabIndex={0}
                  onKeyDown={() => {}}
                  onClick={() => {
                    if (columnCount !== count) {
                      setColumnCount(count);
                    }
                  }}
                >
                  {count}
                </span>
              ))}
            </div>
          </div>
          <div className={css.byline}>
            <div className={css.gamesCount}>
              Game amount:
              <span>{` ${filteredGames.length}`}</span>
            </div>
            <button
              type="button"
              className={css.resetFilters}
              onClick={() => {
                resetFilters();
              }}
            >
              Reset
            </button>
          </div>
        </div>
      </div>
    </div>
  );
};
