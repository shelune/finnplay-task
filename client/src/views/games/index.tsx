import React, { FC, memo } from "react";
import { useData } from "./data";
import { GamesView as MainComponent } from "./games";

export const GamesView: FC = memo(
  () => {
    const props = useData();
    // eslint-disable-next-line react/jsx-props-no-spreading
    return <MainComponent {...props} />;
  },
  () => true,
);
