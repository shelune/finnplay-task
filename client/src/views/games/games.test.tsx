import React from "react";
import { render } from "@testing-library/react";
import {
  mockGameGroups,
  mockGames,
  mockProviders,
} from "../../__mock__/games.mock";
import { GamesView } from "./games";

describe("GamesView", () => {
  it("should render", () => {
    const { container } = render(
      <GamesView
        games={mockGames}
        providers={mockProviders}
        groups={mockGameGroups}
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
