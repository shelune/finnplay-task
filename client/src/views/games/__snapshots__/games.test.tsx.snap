// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GamesView should render 1`] = `
<div>
  <div
    class="gamesPage"
  >
    <div
      class="content"
    >
      <div
        class="gameFiltersMobile"
      >
        <div
          class="filterBySearch"
        >
          <div
            class="searchBar"
          >
            <input
              class="textInput"
              placeholder="Search"
              type="text"
              value=""
            />
            <svg
              class="searchIcon"
            >
              search.svg
            </svg>
          </div>
        </div>
        <div
          class="toggleFilters"
          role="button"
          tabindex="0"
        >
          <svg
            class="toggleFiltersIcon"
          >
            hamburger.svg
          </svg>
          <span>
            Show filters
          </span>
        </div>
      </div>
      <div
        class="gameList"
        style="grid-template-columns: repeat(4, 1fr);"
      >
        <div
          class="gameBox"
          style="background-image: url(https://1540257937.rsc.cdn77.org/desktop/images/games/provider-id-37/GA3Kings.jpg);"
        />
        <div
          class="gameBox"
          style="background-image: url(https://1540257937.rsc.cdn77.org/desktop/images/games/provider-id-37/GA4Symbols.jpg);"
        />
        <div
          class="gameBox"
          style="background-image: url(https://1540257937.rsc.cdn77.org/desktop/images/games/provider-id-7/GT40SurpremeFruits.jpg);"
        />
        <div
          class="gameBox"
          style="background-image: url(https://1540257937.rsc.cdn77.org/desktop/images/games/provider-id-37/CA88Riches.jpg);"
        />
        <div
          class="gameBox"
          style="background-image: url(https://1540257937.rsc.cdn77.org/desktop/images/games/provider-id-37/GAAfricanSunset.jpg);"
        />
        <div
          class="gameBox"
          style="background-image: url(https://1540257937.rsc.cdn77.org/desktop/images/games/provider-id-7/GTAmazingStars.jpg);"
        />
        <div
          class="gameBox"
          style="background-image: url(https://1540257937.rsc.cdn77.org/desktop/images/games/provider-id-37/GAAncientGong.jpg);"
        />
        <div
          class="gameBox"
          style="background-image: url(https://1540257937.rsc.cdn77.org/desktop/images/games/provider-id-7/GTApolloGodOfTheSun10WW.jpg);"
        />
        <div
          class="gameBox"
          style="background-image: url(https://1540257937.rsc.cdn77.org/desktop/images/games/provider-id-37/GAAtlantisWorld.jpg);"
        />
        <div
          class="gameBox"
          style="background-image: url(https://1540257937.rsc.cdn77.org/desktop/images/games/provider-id--3/SWI_Blackjack.jpg);"
        />
        <div
          class="gameBox"
          style="background-image: url(https://1540257937.rsc.cdn77.org/desktop/images/games/provider-id-7/GTBookOfRaClassicLVC.jpg);"
        />
        <div
          class="gameBox"
          style="background-image: url(https://1540257937.rsc.cdn77.org/desktop/images/games/provider-id-7/GTBookofRadeluxe10.jpg);"
        />
        <div
          class="gameBox"
          style="background-image: url(https://1540257937.rsc.cdn77.org/desktop/images/games/provider-id-7/GTBookofRaDeluxe6.jpg);"
        />
        <div
          class="gameBox"
          style="background-image: url(https://1540257937.rsc.cdn77.org/desktop/images/games/provider-id-7/GTBookofRaTempleofGold.jpg);"
        />
        <div
          class="gameBox"
          style="background-image: url(https://1540257937.rsc.cdn77.org/desktop/images/games/provider-id-83/NCBookofShadows.jpg);"
        />
        <div
          class="gameBox"
          style="background-image: url(https://1540257937.rsc.cdn77.org/desktop/images/games/provider-id-37/GABurningFlame.jpg);"
        />
        <div
          class="gameBox"
          style="background-image: url(https://1540257937.rsc.cdn77.org/desktop/images/games/provider-id-37/GACaligula.jpg);"
        />
        <div
          class="gameBox"
          style="background-image: url(https://1540257937.rsc.cdn77.org/desktop/images/games/provider-id-1/MG_PG_CandyBonanza.jpg);"
        />
        <div
          class="gameBox"
          style="background-image: url(https://1540257937.rsc.cdn77.org/desktop/images/games/provider-id-1/MG_PG_CaptainsBounty.jpg);"
        />
        <div
          class="gameBox"
          style="background-image: url(https://1540257937.rsc.cdn77.org/desktop/images/games/provider-id-37/GACastleBlood.jpg);"
        />
        <div
          class="gameBox"
          style="background-image: url(https://1540257937.rsc.cdn77.org/desktop/images/games/provider-id-37/GAChineseZodiac.jpg);"
        />
        <div
          class="gameBox"
          style="background-image: url(https://1540257937.rsc.cdn77.org/desktop/images/games/provider-id-1/MG_PG_CircusDelight.jpg);"
        />
        <div
          class="gameBox"
          style="background-image: url(https://1540257937.rsc.cdn77.org/desktop/images/games/provider-id-37/GACleopatraJewels.jpg);"
        />
        <div
          class="gameBox"
          style="background-image: url(https://1540257937.rsc.cdn77.org/desktop/images/games/provider-id--3/SWI_Cocktailbook.jpg);"
        />
        <div
          class="gameBox"
          style="background-image: url(https://1540257937.rsc.cdn77.org/desktop/images/games/provider-id-101/RG_PG_CocktailNights.jpg);"
        />
        <div
          class="gameBox"
          style="background-image: url(https://1540257937.rsc.cdn77.org/desktop/images/games/provider-id-1/MG_PG_CryptoGold.jpg);"
        />
        <div
          class="gameBox"
          style="background-image: url(https://1540257937.rsc.cdn77.org/desktop/images/games/provider-id-37/GACrystalMystery.jpg);"
        />
        <div
          class="gameBox"
          style="background-image: url(https://1540257937.rsc.cdn77.org/desktop/images/games/provider-id-37/GADancingLion.jpg);"
        />
        <div
          class="gameBox"
          style="background-image: url(https://1540257937.rsc.cdn77.org/desktop/images/games/provider-id-37/GADaVinciCodex.jpg);"
        />
        <div
          class="gameBox"
          style="background-image: url(https://1540257937.rsc.cdn77.org/desktop/images/games/provider-id-97/BL_DiamondJackpots.jpg);"
        />
        <div
          class="gameBox"
          style="background-image: url(https://1540257937.rsc.cdn77.org/desktop/images/games/provider-id-1/MG_PG_DragonHatch.jpg);"
        />
        <div
          class="gameBox"
          style="background-image: url(https://1540257937.rsc.cdn77.org/desktop/images/games/provider-id-37/GADragonLady.jpg);"
        />
        <div
          class="gameBox"
          style="background-image: url(https://1540257937.rsc.cdn77.org/desktop/images/games/provider-id-1/MG_PG_DreamsofMacau.jpg);"
        />
        <div
          class="gameBox"
          style="background-image: url(https://1540257937.rsc.cdn77.org/desktop/images/games/provider-id-1/MG_PG_EgyptBookMystery.jpg);"
        />
        <div
          class="gameBox"
          style="background-image: url(https://1540257937.rsc.cdn77.org/desktop/images/games/provider-id--3/SWI_ElAndaluz.jpg);"
        />
        <div
          class="gameBox"
          style="background-image: url(https://1540257937.rsc.cdn77.org/desktop/images/games/provider-id-37/GAElToreo.jpg);"
        />
        <div
          class="gameBox"
          style="background-image: url(https://1540257937.rsc.cdn77.org/desktop/images/games/provider-id-101/RG_PG_EmojiRiches.jpg);"
        />
        <div
          class="gameBox"
          style="background-image: url(https://1540257937.rsc.cdn77.org/desktop/images/games/provider-id-37/GAEmperorsWealth.jpg);"
        />
        <div
          class="gameBox"
          style="background-image: url(https://1540257937.rsc.cdn77.org/desktop/images/games/provider-id-37/GAExplosiveReels.jpg);"
        />
        <div
          class="gameBox"
          style="background-image: url(https://1540257937.rsc.cdn77.org/desktop/images/games/provider-id--3/SWI_ExtraWinX.jpg);"
        />
        <div
          class="gameBox"
          style="background-image: url(https://1540257937.rsc.cdn77.org/desktop/images/games/provider-id-37/GAFiveStarLuxury.jpg);"
        />
        <div
          class="gameBox"
          style="background-image: url(https://1540257937.rsc.cdn77.org/desktop/images/games/provider-id-93/MERFlamesOfEgypt.jpg);"
        />
        <div
          class="gameBox"
          style="background-image: url(https://1540257937.rsc.cdn77.org/desktop/images/games/provider-id-37/GAFlamingReels.jpg);"
        />
        <div
          class="gameBox"
          style="background-image: url(https://1540257937.rsc.cdn77.org/desktop/images/games/provider-id-37/GAFortuneLions.jpg);"
        />
        <div
          class="gameBox"
          style="background-image: url(https://1540257937.rsc.cdn77.org/desktop/images/games/provider-id-37/GAFortunePanda.jpg);"
        />
        <div
          class="gameBox"
          style="background-image: url(https://1540257937.rsc.cdn77.org/desktop/images/games/provider-id--3/SWI_FruitFever.jpg);"
        />
        <div
          class="gameBox"
          style="background-image: url(https://1540257937.rsc.cdn77.org/desktop/images/games/provider-id-37/GAGoldOfRa.jpg);"
        />
        <div
          class="gameBox"
          style="background-image: url(https://1540257937.rsc.cdn77.org/desktop/images/games/provider-id--3/SWI_GoldenBuffalo.jpg);"
        />
        <div
          class="gameBox"
          style="background-image: url(https://1540257937.rsc.cdn77.org/desktop/images/games/provider-id-37/GAGoldenDragon.jpg);"
        />
        <div
          class="gameBox"
          style="background-image: url(https://1540257937.rsc.cdn77.org/desktop/images/games/provider-id--3/SWI_GoldenPig.jpg);"
        />
        <div
          class="gameBox"
          style="background-image: url(https://1540257937.rsc.cdn77.org/desktop/images/games/provider-id-93/MERHocusPocusDeluxe.jpg);"
        />
        <div
          class="gameBox"
          style="background-image: url(https://1540257937.rsc.cdn77.org/desktop/images/games/provider-id-83/NCIceIceYeti.jpg);"
        />
        <div
          class="gameBox"
          style="background-image: url(https://1540257937.rsc.cdn77.org/desktop/images/games/provider-id-37/GAJadeTreasure.jpg);"
        />
        <div
          class="gameBox"
          style="background-image: url(https://1540257937.rsc.cdn77.org/desktop/images/games/provider-id-37/GAJoanOfArc.jpg);"
        />
        <div
          class="gameBox"
          style="background-image: url(https://1540257937.rsc.cdn77.org/desktop/images/games/provider-id-37/GAJumpinPot.jpg);"
        />
        <div
          class="gameBox"
          style="background-image: url(https://1540257937.rsc.cdn77.org/desktop/images/games/provider-id-7/GTKatana.jpg);"
        />
        <div
          class="gameBox"
          style="background-image: url(https://1540257937.rsc.cdn77.org/desktop/images/games/provider-id-97/BL_KingKongCashpotsJK.jpg);"
        />
        <div
          class="gameBox"
          style="background-image: url(https://1540257937.rsc.cdn77.org/desktop/images/games/provider-id-37/GAKingOfMonkeys.jpg);"
        />
        <div
          class="gameBox"
          style="background-image: url(https://1540257937.rsc.cdn77.org/desktop/images/games/provider-id-37/GAKingOfMonkeys2.jpg);"
        />
        <div
          class="gameBox"
          style="background-image: url(https://1540257937.rsc.cdn77.org/desktop/images/games/provider-id-37/GAKingOfWealth.jpg);"
        />
        <div
          class="gameBox"
          style="background-image: url(https://1540257937.rsc.cdn77.org/desktop/images/games/provider-id-37/GAKittyTwins.jpg);"
        />
        <div
          class="gameBox"
          style="background-image: url(https://1540257937.rsc.cdn77.org/desktop/images/games/provider-id-37/GALadyLuck.jpg);"
        />
        <div
          class="gameBox"
          style="background-image: url(https://1540257937.rsc.cdn77.org/desktop/images/games/provider-id--3/SWI_Leokan.jpg);"
        />
        <div
          class="gameBox"
          style="background-image: url(https://1540257937.rsc.cdn77.org/desktop/images/games/provider-id-37/GALuckyBabies.jpg);"
        />
        <div
          class="gameBox"
          style="background-image: url(https://1540257937.rsc.cdn77.org/desktop/images/games/provider-id-1/MG_PG_LuckyNeko.jpg);"
        />
        <div
          class="gameBox"
          style="background-image: url(https://1540257937.rsc.cdn77.org/desktop/images/games/provider-id-101/RG_PG_LuckyNeko.jpg);"
        />
        <div
          class="gameBox"
          style="background-image: url(https://1540257937.rsc.cdn77.org/desktop/images/games/provider-id-37/GAMagicUnicorn.jpg);"
        />
        <div
          class="gameBox"
          style="background-image: url(https://1540257937.rsc.cdn77.org/desktop/images/games/provider-id-1/MG_PG_MajesticTreasures.jpg);"
        />
        <div
          class="gameBox"
          style="background-image: url(https://1540257937.rsc.cdn77.org/desktop/images/games/provider-id-101/RG_PG_MaskCarnival.jpg);"
        />
        <div
          class="gameBox"
          style="background-image: url(https://1540257937.rsc.cdn77.org/desktop/images/games/provider-id-7/GTMegaJoker.jpg);"
        />
        <div
          class="gameBox"
          style="background-image: url(https://1540257937.rsc.cdn77.org/desktop/images/games/provider-id-37/GAMoneyFarm.jpg);"
        />
        <div
          class="gameBox"
          style="background-image: url(https://1540257937.rsc.cdn77.org/desktop/images/games/provider-id-37/GAMoreCash.jpg);"
        />
        <div
          class="gameBox"
          style="background-image: url(https://1540257937.rsc.cdn77.org/desktop/images/games/provider-id-37/GANightAtKTV.jpg);"
        />
        <div
          class="gameBox"
          style="background-image: url(https://1540257937.rsc.cdn77.org/desktop/images/games/provider-id-101/RG_PG_OrientalProsperity.jpg);"
        />
        <div
          class="gameBox"
          style="background-image: url(https://1540257937.rsc.cdn77.org/desktop/images/games/provider-id-37/GAPhoenixPrincess.jpg);"
        />
        <div
          class="gameBox"
          style="background-image: url(https://1540257937.rsc.cdn77.org/desktop/images/games/provider-id-1/MG_PG_QueenofBounty.jpg);"
        />
        <div
          class="gameBox"
          style="background-image: url(https://1540257937.rsc.cdn77.org/desktop/images/games/provider-id-37/GARamsesTreasure.jpg);"
        />
        <div
          class="gameBox"
          style="background-image: url(https://1540257937.rsc.cdn77.org/desktop/images/games/provider-id-97/BL_RickAndMortyMegaways.jpg);"
        />
        <div
          class="gameBox"
          style="background-image: url(https://1540257937.rsc.cdn77.org/desktop/images/games/provider-id-37/GARoyalGems.jpg);"
        />
        <div
          class="gameBox"
          style="background-image: url(https://1540257937.rsc.cdn77.org/desktop/images/games/provider-id-37/GASantaSFarm.jpg);"
        />
        <div
          class="gameBox"
          style="background-image: url(https://1540257937.rsc.cdn77.org/desktop/images/games/provider-id--3/SWI_SeaGod.jpg);"
        />
        <div
          class="gameBox"
          style="background-image: url(https://1540257937.rsc.cdn77.org/desktop/images/games/provider-id--3/SWI_SinbadsOdyssey.jpg);"
        />
        <div
          class="gameBox"
          style="background-image: url(https://1540257937.rsc.cdn77.org/desktop/images/games/provider-id-37/GASlotOfMoney.jpg);"
        />
        <div
          class="gameBox"
          style="background-image: url(https://1540257937.rsc.cdn77.org/desktop/images/games/provider-id-37/GASpartansLegacy.jpg);"
        />
        <div
          class="gameBox"
          style="background-image: url(https://1540257937.rsc.cdn77.org/desktop/images/games/provider-id-93/MERSpiritsOfTheEast.jpg);"
        />
        <div
          class="gameBox"
          style="background-image: url(https://1540257937.rsc.cdn77.org/desktop/images/games/provider-id-37/GAStarCash.jpg);"
        />
        <div
          class="gameBox"
          style="background-image: url(https://1540257937.rsc.cdn77.org/desktop/images/games/provider-id--3/SWI_StarStretch.jpg);"
        />
        <div
          class="gameBox"
          style="background-image: url(https://1540257937.rsc.cdn77.org/desktop/images/games/provider-id-37/GAStormingFlame.jpg);"
        />
        <div
          class="gameBox"
          style="background-image: url(https://1540257937.rsc.cdn77.org/desktop/images/games/provider-id-1/MG_PG_SushiOishi.jpg);"
        />
        <div
          class="gameBox"
          style="background-image: url(https://1540257937.rsc.cdn77.org/desktop/images/games/provider-id-37/GATexasRangersReward.jpg);"
        />
        <div
          class="gameBox"
          style="background-image: url(https://1540257937.rsc.cdn77.org/desktop/images/games/provider-id-37/GAThunderBird.jpg);"
        />
        <div
          class="gameBox"
          style="background-image: url(https://1540257937.rsc.cdn77.org/desktop/images/games/provider-id-93/MERTizona.jpg);"
        />
        <div
          class="gameBox"
          style="background-image: url(https://1540257937.rsc.cdn77.org/desktop/images/games/provider-id--3/SWI_TokyoDragon.jpg);"
        />
        <div
          class="gameBox"
          style="background-image: url(https://1540257937.rsc.cdn77.org/desktop/images/games/provider-id-1/MG_PG_TreasuresofAztec.jpg);"
        />
        <div
          class="gameBox"
          style="background-image: url(https://1540257937.rsc.cdn77.org/desktop/images/games/provider-id-93/MERTripleFlame.jpg);"
        />
        <div
          class="gameBox"
          style="background-image: url(https://1540257937.rsc.cdn77.org/desktop/images/games/provider-id-93/MERTropicalHeat.jpg);"
        />
        <div
          class="gameBox"
          style="background-image: url(https://1540257937.rsc.cdn77.org/desktop/images/games/provider-id-30/endTwerk.jpg);"
        />
        <div
          class="gameBox"
          style="background-image: url(https://1540257937.rsc.cdn77.org/desktop/images/games/provider-id-30/endVoodoo.jpg);"
        />
        <div
          class="gameBox"
          style="background-image: url(https://1540257937.rsc.cdn77.org/desktop/images/games/provider-id-93/MERVoodooShark.jpg);"
        />
        <div
          class="gameBox"
          style="background-image: url(https://1540257937.rsc.cdn77.org/desktop/images/games/provider-id-101/RG_PG_WildBandito.jpg);"
        />
        <div
          class="gameBox"
          style="background-image: url(https://1540257937.rsc.cdn77.org/desktop/images/games/provider-id-1/MG_PG_WildFireworks.jpg);"
        />
        <div
          class="gameBox"
          style="background-image: url(https://1540257937.rsc.cdn77.org/desktop/images/games/provider-id-93/MERWildSupreme.jpg);"
        />
        <div
          class="gameBox"
          style="background-image: url(https://1540257937.rsc.cdn77.org/desktop/images/games/provider-id-97/BL_WishUponAJackpot.jpg);"
        />
        <div
          class="gameBox"
          style="background-image: url(https://1540257937.rsc.cdn77.org/desktop/images/games/provider-id-37/GAWolfQuest.jpg);"
        />
        <div
          class="gameBox"
          style="background-image: url(https://1540257937.rsc.cdn77.org/desktop/images/games/provider-id-93/MERZentaurus.jpg);"
        />
      </div>
      <div
        class="gameFilters"
      >
        <div
          class="filterBySearch"
        >
          <div
            class="searchBar"
          >
            <input
              class="textInput"
              placeholder="Search"
              type="text"
              value=""
            />
            <svg
              class="searchIcon"
            >
              search.svg
            </svg>
          </div>
        </div>
        <div
          class="filterByProviders"
        >
          <div
            class="filterTitle"
          >
            Providers
          </div>
          <div
            class="filterList"
          >
            <span
              class="filterPill"
              role="button"
              tabindex="0"
            >
              Swintt
            </span>
            <span
              class="filterPill"
              role="button"
              tabindex="0"
            >
              Microgaming
            </span>
            <span
              class="filterPill"
              role="button"
              tabindex="0"
            >
              Greentube
            </span>
            <span
              class="filterPill"
              role="button"
              tabindex="0"
            >
              Endorphina
            </span>
            <span
              class="filterPill"
              role="button"
              tabindex="0"
            >
              GameArt
            </span>
            <span
              class="filterPill"
              role="button"
              tabindex="0"
            >
              Nolimit City
            </span>
            <span
              class="filterPill"
              role="button"
              tabindex="0"
            >
              Merkur
            </span>
            <span
              class="filterPill"
              role="button"
              tabindex="0"
            >
              Blueprint Gaming
            </span>
            <span
              class="filterPill"
              role="button"
              tabindex="0"
            >
              Relax Gaming
            </span>
          </div>
        </div>
        <div
          class="filterByGroups"
        >
          <div
            class="filterTitle"
          >
            Game groups
          </div>
          <div
            class="filterList"
          >
            <span
              class="filterPill"
              role="button"
              tabindex="0"
            >
              Jackpot
            </span>
            <span
              class="filterPill"
              role="button"
              tabindex="0"
            >
              Popular
            </span>
            <span
              class="filterPill"
              role="button"
              tabindex="0"
            >
              Recent
            </span>
            <span
              class="filterPill"
              role="button"
              tabindex="0"
            >
              Slots
            </span>
          </div>
        </div>
        <div
          class="sorting"
        >
          <div
            class="filterTitle"
          >
            Sorting
          </div>
          <div
            class="filterList"
          >
            <span
              class="filterPill isActive"
              role="button"
              tabindex="0"
            >
              A-Z
            </span>
            <span
              class="filterPill"
              role="button"
              tabindex="0"
            >
              Z-A
            </span>
            <span
              class="filterPill"
              role="button"
              tabindex="0"
            >
              Newest
            </span>
          </div>
        </div>
        <div
          class="filterByColumns"
        >
          <div
            class="filterTitle"
          >
            Columns
          </div>
          <div
            class="filterBar"
          >
            <div
              class="bar"
            >
              <div
                class="barBackground"
              />
              <div
                class="barActive"
              />
            </div>
            <span
              class="milestone isActive"
              role="button"
              tabindex="0"
            >
              2
            </span>
            <span
              class="milestone isActive"
              role="button"
              tabindex="0"
            >
              3
            </span>
            <span
              class="milestone isActive"
              role="button"
              tabindex="0"
            >
              4
            </span>
          </div>
        </div>
        <div
          class="byline"
        >
          <div
            class="gamesCount"
          >
            Game amount:
            <span>
               105
            </span>
          </div>
          <button
            class="resetFilters"
            type="button"
          >
            Reset
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
